### ISSUE TITLE
Plugin does not work with react props

### ISSUE THREAD

**Issue by jspradlin1-chwy** [JaroslawPokropinski/eslint-plugin-strict-null-checks#24]
this code should show an error in TestComponent where it tries to pass the nullable someProp on to the child component, where it is not nullable:

```tsx
const ChildComponent = ({ someProp }: { someProp: SomeType }) => {
  return <>{someProp.name}</>;
};

const TestComponent = ({ someProp }: { someProp: SomeType | null }) => {
  return <ChildComponent someProp={someProp} />;
};
```

It should be noted that using a strictNullChecks in the global tsconfig does catch this issue, but since we're attempting to migrate incrementally, that won't be a good solution for us.

**Comment by JaroslawPokropinski** [comment]
Hi, can you let me know it the issue still exists? I recently updated the error detection to use typescript errors directly and it might have solved the issue.

**Comment by jspradlin1-chwy** [comment]
The latest version gives a warning, but the message is 'object Object'

![Screenshot 2023-10-11 at 8 50 41 AM](https://github.com/JaroslawPokropinski/eslint-plugin-strict-null-checks/assets/87656333/ca54dcc2-a40e-4794-b402-ce4aa062d72b)

**Comment by jspradlin1-chwy** [comment]
Works fine when it's not a prop:
![Screenshot 2023-10-11 at 8 53 54 AM](https://github.com/JaroslawPokropinski/eslint-plugin-strict-null-checks/assets/87656333/6819eb67-7a52-4f1f-af60-f8b2db179471)

**Comment by jspradlin1-chwy** [comment]
From the CLI:
<img width="703" alt="Screenshot 2023-10-11 at 8 59 38 AM" src="https://github.com/JaroslawPokropinski/eslint-plugin-strict-null-checks/assets/87656333/2a306451-1234-4268-8461-3dc332b6e3c6">
